include/seqan3/io/sequence_file/format_fasta.hpp

```diff
@@ -329,47 +326,35 @@ class format_fasta
         if (options.fasta_blank_before_id)
             stream_it = ' ';
 
-        std::ranges::copy(id, stream_it);
+        stream_it.write_range(id);
 
         detail::write_eol(stream_it, options.add_carriage_return);
     }
 
     //!\brief Implementation of writing the sequence.
-    template <typename stream_it_t,
-              typename seq_type>
-    void write_seq(stream_it_t & stream_it,
-                   sequence_file_output_options const & options,
-                   seq_type && seq)
+    template <typename stream_it_t, typename seq_type>
+    void write_seq(stream_it_t & stream_it, sequence_file_output_options const & options, seq_type && seq)
     {
+        // using distance_type = decltype(std::ranges::distance(seq));
+
+        auto v = seq | views::to_char;
+
         if (options.fasta_letters_per_line > 0)
         {
-        #if SEQAN3_WORKAROUND_VIEW_PERFORMANCE
-            size_t count = 0;
-            for (auto c : seq)
+            size_t written_characters{0};
+            auto it = v.begin();
+
+            while (it != v.end())
             {
-                stream_it = to_char(c);
-                if (++count % options.fasta_letters_per_line == 0)
-                    detail::write_eol(stream_it, options.add_carriage_return);
-            }
-            if (count % options.fasta_letters_per_line != 0)
+                it = stream_it.write_range(v | views::drop(written_characters)
+                                             | views::take(options.fasta_letters_per_line));
                 detail::write_eol(stream_it, options.add_carriage_return);
-
-        #else // ↑↑↑ WORKAROUND | ORIGINAL ↓↓↓
-
-            //TODO: combining chunk and join is substantially faster than views::interleave (2.5x), why?
-            std::ranges::copy(seq | views::to_char
-                                  | ranges::view::chunk(options.fasta_letters_per_line)
-                                  | views::join(options.add_carriage_return
-                                                    ? std::string_view{"\r\n"}
-                                                    : std::string_view{"\n"}),
-                              stream_it);
-            detail::write_eol(stream_it, options.add_carriage_return);
-        #endif // SEQAN3_WORKAROUND_VIEW_PERFORMANCE
+                written_characters += options.fasta_letters_per_line;
+            }
```

Comment:
Ok, I know I said we don't care about non-sized or non-random-access ranges. BUT :rofl: 

views::drop is has linear construction on non-random-access ranges in regard to its parameter. And you are going over the whole sequence again every iteration, first the first 70 chars, next time 140 chars... Although you in fact have the iterator where the sequence starts every time! I propose the following loop which is almost identical:

```cpp
            while (it != v.end())
            {
                 auto e = it;
                 std::ranges::advance(e, options.fasta_letters_per_line, std::ranges::end(v));
                 auto r = std::ranges::subrange{it, e};
                 it = stream_it.write_range(r);
            }
```